### ISSUE TITLE
Path value with a string containing a numeric value followed by a space is incorrectly parsed

### ISSUE THREAD

**Issue by benmj** [Netflix/falcor#785]
Part of my path is the string "30 " (note the trailing space). This is being incorrectly parsed by Falcor to the server as a numeric value _30_.

**Comment by jhusain** [comment]
Is it "30 " when sent to the server or 30? The route attempting to parse {integers} Or {keys}?

**Comment by sdesai** [comment]
30 (no space, a number). The route is expecting keys.

**Comment by sdesai** [comment]
Should have clarified, this is in the context of a search (hence keys) - for example backspacing from "30 Ro"

**Comment by greim** [comment]
Possibly related?

https://github.com/Netflix/falcor-router/issues/176
https://github.com/Netflix/falcor-path-utils/pull/6

**Comment by sdesai** [comment]
Looks like it's coming from here:
https://github.com/Netflix/falcor-path-utils/blob/master/lib/toPaths.js#L67

When batching paths for the request.

However, I'd imagine it's too late here, to distinguish between:

```
"foo" : {
   0: {"bar": {}},
}
```

And

```
"foo" : {
   "30": { "bar": {} },
}
```

As the map representation for ["foo", 0, "bar" ...] and ["foo", "30", "bar"] respectively (since they both get coerced to strings when added/retrieved as keys). I imagine if we drop the parseInt(), then we aren't in a position to collapse continuous keys into ranges.

If we need to fix this, it may need to come from changes higher in the call stack (GetRequest.batch -> flushGetRequest -> toPaths)

**Comment by asyncanup** [comment]
This is no longer an issue, fixed by falcor-path-utils v0.5.1
Confirmed by running a falcor-http-datasource locally and logging queries:
```
$ node server.js
url /model.json?paths=%5B%5B%22videos%22%2C%2230%20%22%2C%22title%22%5D%5D&method=get
query { paths: '[["videos","30 ","title"]]', method: 'get' }
```

Closing. @benmj please re-open if this is still an issue.
